Add `Daily_challenge` (#12392)
ppy/osu-wiki: wiki/Gameplay/Daily_challenge/en.md

# Daily challenge

> The **daily challenge** is a multiplayer mode in osu!(lazer) where players are able to build up a daily streak by passing beatmaps consecutively with increasing difficulty with the difficulty being reset every 7 days.

Reviewer: ```suggestion
The **daily challenge** is a multiplayer mode in [osu!(lazer)](/wiki/Client/Release_stream/Lazer) where players are able to build up a daily streak by passing beatmaps consecutively with increasing difficulty with the difficulty being reset every 7 days.
```

> The **daily challenge** is a multiplayer mode in [osu!(lazer)](/wiki/Client/Release_stream/Lazer) where players are able to build up a daily streak by passing beatmaps consecutively with increasing difficulty with the difficulty being reset every 7 days.

Reviewer 2: ```suggestion
The **daily challenge** is a multiplayer mode in [osu!(lazer)](/wiki/Client/Release_stream/Lazer) where players are able to build up a streak by passing consecutive daily beatmaps with an increasing difficulty that resets every 7 days.
```

> # Daily Challenge

Reviewer 3: ```suggestion
# Daily challenge
```

The **daily challenge** is a multiplayer mode in osu!(lazer) where players are able to build up a daily streak by passing beatmaps consecutively with increasing difficulty with the difficulty being reset every 7 days.

> Every beatmap comes from the [Featured Artist](/wiki/People/Featured_Artists) listing, and sometimes the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.

Author: Yea it also makes more sense since they are handpicked and not... idk, picked randomly by some sort of algorythm?
Will change rn.

> Every beatmap is handpicked  from the [Featured Artist](/wiki/People/Featured_Artists) listing, and sometimes the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.

Reviewer 4: ```suggestion
Every beatmap is handpicked from the [Featured Artist](/wiki/People/Featured_Artists) listing, and sometimes the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.
```

> Every beatmap is handpicked  from the [Featured Artist](/wiki/People/Featured_Artists) listing, and sometimes the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.

Reviewer 4: ```suggestion
Every beatmap is handpicked from the [Featured Artist](/wiki/People/Featured_Artists) listing, and sometimes the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.
```

## Game menu

From the main menu, the daily challenge menu can be accessed with the following steps:

> 1. Click the `Play` button or press `P`.

Reviewer 3: ```suggestion
1. Click the `play` button or press `P`.
```

> 2. Click the `Daily challenge` button or press `D`.

Reviewer 3: ```suggestion
2. Click the `daily challenge` button or press `D`.
```

![](img/daily-challenge-menu.png "Screenshot of the daily challenge menu during November 7, 2024.") 

On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.

After the intro ends, the user can view the data for today's challenge. The left side displays various score-related information like the total pass count and the cumulative total score. In the middle, a leaderboard shows the top scores achieved by players. Users are able to discuss the daily challenge in the chat on the right.

## Beatmap difficulty range

![](img/Beatmap-selection-criteria.png "Specific star ratings beatmaps have to follow for the entire week")

> 

Author: I'm not sure what else could be added.
I could make it into a table tho.
I'll keep it the way it is for now, but will change if you insist again (It doesn't really matter to me tbh, just not sure if there is a reason to for now)

> 

Reviewer 3: I think it's fine as-is.

## Streak milestones

Play streaks are displayed on the user profile with different colours depending on certain streak milestones.

> ### Rank breakdown 

Reviewer 3: would be superb if you showcase the colors in the table.

what I'm thinking is having a screenshot of the daily challenge streak text in each row with its respective color.

should be pretty straightforward to make with some HTML editing. let me know if you need help with this otherwise.

Author: > should be pretty straightforward to make with some HTML editing.

I'm too bad with JS and using commands in the console overall to do this, so I would appreciate some help 😅 (I'm new to even HTML, so don't judge me *too* much)

Tho in what way are you planning to show the colours? Are you going to input the day/week requirement in the console and later replace the requirement text with the screenshot, or do you want to do this for only a specific type of streak, or an even more different way?
I feel like replacing the text with the screenshots completely would feel tacky, but we can try that anyway I suppose.

Reviewer 3: made this, can send over my changes if you want.

![image](https://github.com/user-attachments/assets/9edbffcb-08c7-4d56-9f7b-23f54291ba04)

Reviewer 5: It's weird they are all offset by one day. That wasn't the intention. If this is an error with implementation we should fix that first?

Reviewer 4: it's just this line: https://github.com/ppy/osu-web/blob/b24f634812725d35bf720f344f58536990360022/resources/js/profile-page/daily-challenge.tsx#L25

Reviewer 5: well we either change that or change this page to read `>1080` etc. it looks so weird atm

Reviewer 3: I've looked into this on lazer/web, and adjusting the checks in https://github.com/ppy/osu/blob/master/osu.Game/Overlays/Profile/Header/Components/DailyChallengeStatsTooltip.cs#L148 to use `>=` for the checks would result in even weirder numbers for total participation and weekly streak because of *rounding*, such as 1079d counting as Lustrous in total participation.

Would be better to just use `>1080` here for now until we do more consideration for the current implementation.

Reviewer 5: I don't see why rounding should come into the equation. That sounds like a bug regardless of the off-by-one consideration?

Reviewer 3: I've opened PRs to attempt having expected and consistent results across web and client for this, while explaining the issue there:

https://github.com/ppy/osu-web/pull/11698
https://github.com/ppy/osu/pull/30906

Author: since one of the pr's was already merged, I'm assuming we just change the requirements in the table to match the new calculation.
Tho, do you still want me to add a `>` to the numbers? I can't see a reason to do that, but id rather ask than to push changes already.

Author: will apply with the new batch of changes, closing.

> | Rank | Total streak | Daily streak | Weekly streak |

Reviewer 4: ```suggestion
| Rank | Total participation | Daily streak | Weekly streak |
```

> | Rank | Total streak | Daily streak | Weekly streak |
> | :-: | :-: | :-: | :-: |
> | Lustrous | 1081 days | 360 days | 53 weeks |
> | Radiant | 721 days | 240 days | 36 weeks |
> | Rhodium | 361 days | 120 days | 19 weeks |
> | Platinum | 181 days | 60 days | 10 weeks |
> | Gold | 91 days | 30 days | 6 weeks |
> | Silver | 31 days | 10 days | 3 weeks |
> | Bronze | 16 days | 5 days | 2 weeks |
> | Iron | less than 16 days | less than 5 days | less than 2 weeks |

Reviewer 4: ```suggestion
| Tier | Total participation | Daily streak | Weekly streak |
| :-: | :-: | :-: | :-: |
| Lustrous | 1081 days | 361 days | 53 weeks |
| Radiant | 721 days | 241 days | 36 weeks |
| Rhodium | 361 days | 121 days | 19 weeks |
| Platinum | 181 days | 61 days | 10 weeks |
| Gold | 91 days | 31 days | 6 weeks |
| Silver | 31 days | 11 days | 3 weeks |
| Bronze | 16 days | 6 days | 2 weeks |
| Iron | less than 16 days | less than 6 days | less than 2 weeks |
```

## Contributors

> 

Reviewer 3: using a table feels a bit too repetitive here, would suggest doing the following:

```md
The project is organized by ::{ flag=TN }:: [Hivie](https://osu.ppy.sh/users/14102976). The following community members are responsible for picking beatmaps:

- ::{ flag=US }:: [ChillierPear](https://osu.ppy.sh/users/9501251)
- ::{ flag=BR }:: [Dada](https://osu.ppy.sh/users/9501251)
- ::{ flag=DE }:: [FuJu](https://osu.ppy.sh/users/10773882)
- ::{ flag=KR }:: [momoyo](https://osu.ppy.sh/users/12469536)
- ::{ flag=GB }:: [kingling](https://osu.ppy.sh/users/7010761)
- ::{ flag=KR }:: [Luscent](https://osu.ppy.sh/users/2688581)
```

this also includes an update to the user listing (DeviousPanda -> momoyo)

| Name | Role |
| :-- | :-- |
| ::{ flag=TN }:: [Hivie](https://osu.ppy.sh/users/14102976) | Organisation effort lead |
| ::{ flag=US }:: [ChillierPear](https://osu.ppy.sh/users/9501251) | Beatmap picking |
| ::{ flag=BR }:: [Dada](https://osu.ppy.sh/users/9501251) | Beatmap picking |
| ::{ flag=DE }:: [FuJu](https://osu.ppy.sh/users/10773882) | Beatmap picking |
| ::{ flag=GB }:: [DeviousPanda](https://osu.ppy.sh/users/4966334) | Beatmap picking |
| ::{ flag=GB }:: [kingling](https://osu.ppy.sh/users/7010761) | Beatmap picking |
| ::{ flag=KR }:: [Luscent](https://osu.ppy.sh/users/2688581) | Beatmap picking |

## Trivia

::: Infobox
![](img/yt-comment.jpg "waxxx14's YouTube comment suggesting the \"cup of the day\" format")
:::

> - The idea for daily challenge was taken from a comment from waxxx14 requesting a "cup of the day" format from [TrackMania](https://en.wikipedia.org/wiki/TrackMania) in osu!, under a lazer dev video, ["deciding what to do with lazer"](https://www.youtube.com/watch?v=xUSxEjQQ1UI)

Author: i think development makes more sense since even tho it was an update video, it also served a purpose to get community feedback on what should be the next main focus for laser + it just sounds nicer IMO.

> - The idea for daily challenge was taken from a comment from waxxx14 requesting a "cup of the day" format from [TrackMania](https://en.wikipedia.org/wiki/TrackMania) in osu!, under a lazer dev video, ["deciding what to do with lazer"](https://www.youtube.com/watch?v=xUSxEjQQ1UI)

Reviewer 6: Yeah sure :thumbs_up:

> - The idea for daily challenge was taken from a comment from waxxx14 requesting a "cup of the day" format from [TrackMania](https://en.wikipedia.org/wiki/TrackMania) in osu!, posted under the lazer development video ["deciding what to do with lazer"](https://www.youtube.com/watch?v=xUSxEjQQ1UI)

Reviewer 3: ```suggestion
- The idea for daily challenge was taken from a comment from waxxx14 requesting a "cup of the day" format from [TrackMania](https://en.wikipedia.org/wiki/TrackMania) in osu!, posted under the lazer development video ["deciding what to do with lazer"](https://www.youtube.com/watch?v=xUSxEjQQ1UI).
```

## Game menu

From the main menu, the daily challenge menu can be accessed with the following steps:

1. Click the `play` button or press `P`.
2. Click the `daily challenge` button or press `D`.

![](img/daily-challenge-menu.png "Screenshot of the daily challenge menu during November 7, 2024.") 

On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.

After the intro ends, the user can view the data for today's challenge. The left side displays various score-related information like the total pass count and the cumulative total score. In the middle, a leaderboard shows the top scores achieved by players. Users are able to discuss the daily challenge in the chat on the right.

## Beatmap difficulty range

![](img/Beatmap-selection-criteria.png "Specific star ratings beatmaps have to follow for the entire week")

## Streak milestones

Play streaks are displayed on the user profile with different colours depending on certain streak milestones.

### Tier breakdown

<!-- tier images: https://www.figma.com/design/tc79qAgJ35KQvdTO0Oj3dN/Daily-Challenge-Counter?node-id=0-1&t=xjRm9Ke0tUMtAQlh-1 -->

> | | Tier | Total participation | Daily streak | Weekly streak |

Reviewer 6: ```suggestion
|  | Tier | Total participation | Daily streak | Weekly streak |
```

This is an oversight and not due to the images, right?

Reviewer 3: it's an intentional choice to make the images properly aligned in their own column, we've done this kind of thing before iirc

Reviewer 3: oh yea it's causing a CI fail because of the missing space, will fix

# Daily Challenge

The **daily challenge** is a multiplayer mode in osu!(lazer) where players are able to build up a *daily* streak by passing beatmaps consecutively with increasing difficulty.

Each day introduces one new map from the [Featured Artist](/wiki/People/Featured_Artists) listing with increased difficulty. Once every 7 days however, the difficulty resets again.

Sometimes, the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.

## Game menu

From the main menu, the daily challenge menu can be accessed with the following steps:

1. Click the `Play` button or press `P`.
2. Click the `Daily challenge` button or press `D`.

> ![](img/Daily-challenge-menu.png "Screenshot of the daily challenge menu during November 7, 2024.") 

Reviewer 6: ```suggestion
![](img/daily-challenge-menu.png "Screenshot of the daily challenge menu during November 7, 2024.") 
```

The image is not displayed correctly because the name doesn't match.

Author: It works for me (i have a plugin in VS code that allows me to preview the article in osu! wiki styling)
Will change anyway because it could really work incorrectly in other places.

On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.

After the intro ends, the user can view the data for today's challenge. The left side displays various score-related information like the total pass count and the cumulative total score. In the middle, a leaderboard shows the top scores achieved by players. Users are able to discuss the daily challenge in the chat on the right.

## Beatmap difficulty range

![](img/Beatmap-selection-criteria.png "Specific star ratings beatmaps have to follow for the entire week")

## Streak milestones

Play streaks are displayed on the user profile with different colours depending on certain streak milestones.

> #### Rank breakdown 

Reviewer 6: ```suggestion
### Rank breakdown
```

> | Rank  | Total streak | Daily streak | Weekly streak |

Reviewer 6: ```suggestion
| Rank | Total streak | Daily streak | Weekly streak |
```

| :-: | :-: | :-: | :-: |

> | Lustrous  | 1081 days | 360 days | 53 weeks |

Reviewer 6: ```suggestion
| Lustrous | 1081 days | 360 days | 53 weeks |
```

| Radiant | 721 days | 240 days | 36 weeks |

> | Rhodium | 361 days| 120 days | 19 weeks |

Reviewer 6: ```suggestion
| Rhodium | 361 days | 120 days | 19 weeks |
```

| Platinum | 181 days | 60 days | 10 weeks |
| Gold | 91 days | 30 days | 6 weeks |

> | Silver | 31 days | 10 days | 3 weeks  |

Reviewer 6: ```suggestion
| Silver | 31 days | 10 days | 3 weeks |
```

| Bronze | 16 days | 5 days | 2 weeks |

> | Iron | less than 16 days | less than 5 days | less than 2 weeks |--->

Reviewer 6: ```suggestion
| Iron | less than 16 days | less than 5 days | less than 2 weeks |
```

Author: must have left that after deleting the comment, oops, my bad

## Contributors

| Name | Role |
| :-- | :-- |
| ::{ flag=TN }:: [Hivie](https://osu.ppy.sh/users/14102976) | Organisation effort lead |

> | ::{ flag=US }:: [ChillierPear](https://osu.ppy.sh/users/9501251) | Beatmap picking

Reviewer 6: ```suggestion
| ::{ flag=US }:: [ChillierPear](https://osu.ppy.sh/users/9501251) | Beatmap picking |
```
